packages/ckeditor5-emoji/src/emojirepository.ts

```diff
@@ -226,11 +239,25 @@ export default class EmojiRepository extends Plugin {
 	}
 
 	/**
-	 * A function used to check if the given emoji is supported in the operating system.
+	 * A function used to check if the given ZWJ emoji is supported in the operating system.
+	 *
+	 * Referenced for unit testing purposes.
+	 */
+	private static _isEmojiZwjSupported = isEmojiZwjSupported;
+
+	/**
+	 * A function used to get the emoji version supported by the operating system.
+	 *
+	 * Referenced for unit testing purposes.
+	 */
+	private static _getEmojiSupportedVersionByOs = getEmojiSupportedVersionByOs;
+
+	/**
+	 * A function used to determine if emoji has a zero width joiner.
 	 *
 	 * Referenced for unit testing purposes.
 	 */
-	private static _isEmojiSupported = isEmojiSupported;
+	private static _hasZwj = hasZwj;
```

Comment:
I would think a bit if we could create an `EmojiUtils` plugin to move internals to a dedicated object that could be checked (via tests) but also gives the option to create some mocks in `EmojiRepository`.

Reply:
I found two direct ways of creating utils files in CKEditor 5 codebase:
1. Exporting each util as a separate function.
2. Creating a util class that holds all required functions.

What solution would you think is best. I guess 2. is easier to mock, but requires more code, and effort to write.

Reply:
`EmojiUtils` as a plugin, so (2).

Reply:
Introduced `EmojiUtils` in Fixed in [7343da3](https://github.com/ckeditor/ckeditor5/pull/17864/commits/7343da3b2734aaf3d93b6e1e8e455c9a2bf881df).

Unfortunately our tests do not support mocking a plugin that requires other plugins. `EmojiPlugin` mocks `EmojiRepository` that requires `EmojiUtils`. Maybe there is a way of solving this, that I'm not aware of, so I'm leaving this till tomorrow.